compiler/dyno/include/chpl/util/subprocess.h

```diff
@@ -48,7 +52,38 @@ int executeAndWait(const std::vector<std::string>& commandVec,
                    const std::string& description,
                    bool printSystemCommands = false);
 
-#endif
+/**
+  Run a command and collect its output into a string.
 
+  command
+    the command to run
+
+  returns
+    an error code if something went wrong, or a string
+    containing the command's output.
+ */
+llvm::ErrorOr<std::string> getCommandOutput(const std::string& command);
+
+using ChplEnvMap = std::unordered_map<std::string, std::string>;
+
+/**
+  Run printchplenv and collect its output into a map.
+
+  varMap
+    additional environment variables to feed to printchplenv
+
+  chplHome
+    the CHPL_HOME environment variable, to locate printchplenv
+
+  returns
+    an error code if something went wrong while running printchplenv,
+    or a map containing key-value pairs for each variable in
+    printchplenv's output.
+ */
+llvm::ErrorOr<ChplEnvMap>
+getChplEnv(const std::map<std::string, const char*>& varMap,
+           const char* chplHome);
```

Comment:
Hmm... this one (and the related `include <unordered_map>` etc) seem unrelated to subprocesses in general. Can we make a new header like maybe `chplenv.h` or something?

I thought for a minute that you could make it a `static` helper method inside of `Context.cpp`, but that seems harder with the production compiler calling it.